acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/publickeys/PublicKeysAcceptanceTestBase.java

```diff
@@ -28,52 +29,117 @@
-    return new String[] {PRIVATE_KEY_1, PRIVATE_KEY_2};
+  protected String[] privateKeys(final String keyType) {
+    return keyType.equals("BLS")
+        ? new String[] {BLS_PRIVATE_KEY_1, BLS_PRIVATE_KEY_2}
+        : new String[] {SECP_PRIVATE_KEY_1, SECP_PRIVATE_KEY_2};
   }
 
-  protected String[] createKeys(boolean isValid, final String... privateKeys) {
+  protected String[] createKeys(
+      final String keyType, boolean isValid, final String... privateKeys) {
+    return keyType.equals("BLS")
+        ? createBlsKeys(isValid, privateKeys)
+        : createSecpKeys(isValid, privateKeys);
+  }
+
+  protected String[] createBlsKeys(boolean isValid, final String... privateKeys) {
     return Stream.of(privateKeys)
         .map(
             privateKey -> {
               final BLSKeyPair keyPair =
                   new BLSKeyPair(BLSSecretKey.fromBytes(Bytes.fromHexString(privateKey)));
-              final Path keyConfigFile = configFileName(keyPair.getPublicKey());
+              final Path keyConfigFile = blsConfigFileName(keyPair.getPublicKey());
               if (isValid) {
                 metadataFileHelpers.createUnencryptedYamlFileAt(keyConfigFile, privateKey);
               } else {
-                try {
-                  Files.createFile(keyConfigFile);
-                } catch (final IOException e) {
-                  throw new UncheckedIOException(e);
-                }
+                createInvalidFile(keyConfigFile);
               }
               return keyPair.getPublicKey().toString();
             })
         .toArray(String[]::new);
   }
 
-  private Path configFileName(final BLSPublicKey publicKey) {
-    final String configFilename2 = publicKey.toString().substring(2);
-    return testDirectory.resolve(configFilename2 + ".yaml");
+  protected String[] createSecpKeys(boolean isValid, final String... privateKeys) {
+    return Stream.of(privateKeys)
+        .map(
+            privateKey -> {
+              final ECKeyPair ecKeyPair =
+                  ECKeyPair.create(Numeric.toBigInt(Bytes.fromHexString(privateKey).toArray()));
+              final String publicKey = Numeric.toHexStringWithPrefix(ecKeyPair.getPublicKey());
+              if (isValid) {
+                createSecpKey(privateKey);
+              } else {
+                final Path keyConfigFile = testDirectory.resolve(publicKey + ".toml");
+                createInvalidFile(keyConfigFile);
+              }
+              return publicKey;
+            })
+        .toArray(String[]::new);
+  }
+
+  private void createInvalidFile(final Path keyConfigFile) {
+    try {
+      Files.createFile(keyConfigFile);
+    } catch (final IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private void createSecpKey(final String blsPrivateKey) {
+    final Bytes privateKey = Bytes.fromHexString(blsPrivateKey);
+    final ECKeyPair ecKeyPair = ECKeyPair.create(Numeric.toBigInt(privateKey.toArray()));
```

Comment:
this was already a bit done in the previous function - do we have to do it again?

Reply:
done